src/main/badge.ts

```diff
@@ -2,20 +2,79 @@
 // Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {app} from 'electron';
+import {BrowserWindow, app, nativeImage} from 'electron';
 import log from 'electron-log';
 
 import {UPDATE_BADGE} from 'common/communication';
 
 import {localizeMessage} from 'main/i18nManager';
 
-import WindowManager from './windows/windowManager';
 import * as AppState from './appState';
+import MainWindow from './windows/mainWindow';
 
 const MAX_WIN_COUNT = 99;
 
 let showUnreadBadgeSetting: boolean;
 
+/**
+     * Badge generation for Windows
+     */
+
+function drawBadge(text: string, small: boolean) {
+    const scale = 2; // should rely display dpi
+    const size = (small ? 20 : 16) * scale;
+    const canvas = document.createElement('canvas');
+    canvas.setAttribute('width', `${size}`);
+    canvas.setAttribute('height', `${size}`);
+    const ctx = canvas.getContext('2d');
+
+    if (!ctx) {
+        log.error('Could not create canvas context');
+        return null;
+    }
+
+    // circle
+    ctx.fillStyle = '#FF1744'; // Material Red A400
+    ctx.beginPath();
+    ctx.arc(size / 2, size / 2, size / 2, 0, Math.PI * 2);
+    ctx.fill();
+
+    // text
+    ctx.fillStyle = '#ffffff';
+    ctx.textAlign = 'center';
+    ctx.textBaseline = 'middle';
+    ctx.font = (11 * scale) + 'px sans-serif';
+    ctx.fillText(text, size / 2, size / 2, size);
+
+    return canvas.toDataURL();
+}
+
+function createDataURL(win: BrowserWindow, text: string, small: boolean) {
+    // since we don't have a document/canvas object in the main process, we use the webcontents from the window to draw.
+    const safeSmall = Boolean(small);
```

Comment:
what is the reason for this "safety"?

Reply:
Not sure, might have been a reason in the past. Will remove for now.